staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -862,7 +862,8 @@ func UpdateResource(r rest.Updater, scope RequestScope, typer runtime.ObjectType
 		defaultGVK := scope.Kind
 		original := r.New()
 		trace.Step("About to convert to expected version")
-		obj, gvk, err := scope.Serializer.DecoderToVersion(s.Serializer, defaultGVK.GroupVersion()).Decode(body, &defaultGVK, original)
+		decoder := scope.Serializer.DecoderToVersion(s.Serializer, schema.GroupVersion{Group: defaultGVK.Group, Version: runtime.APIVersionInternal})
```

Comment:
> How do we know the object in RawExtension is of group defaultGVK.Group?

We don't, but both create and update already make this assumption, so this doesn't make it worse. This is just harmonizing update to behave the same as create, since the intent of the API server is to deal with internal objects.